### ISSUE TITLE
Bayes Factor calculation at the unit level

### ISSUE THREAD

**Issue by pekjoonwu** [stephenslab/mashr#96]
Hi,

Thanks for developing this great tool for analyzing the data. I have a question more about comparing the results from mashr to the other methods such as Metasoft.

I am wondering when you compared the power in Figure 2 a, b panel with Metasoft, were you using the Bayes factor to test the global null so that both methods have significance measurement for unit level? If so, is there any function in the package that can output this factor for each unit and what threshold should be used for each unit?

Thank you so much!

Best

**Comment by stephens999** [comment]
yes that is what we did.

I notice that our documentation of the mash output is quite incomplete!
The log BF can be directly computed from the mash output (res.mash) by
res.mash$alt_loglik - res.mash$null_loglik
I believe this is log base e.

The threshold is up to you... not that easy to give a general answer to that.

**Comment by pekjoonwu** [comment]
Thank you Matthew for the clarification! 

Since I am not a Bayesian, I still have one question about the procedure. Specifically, did you guys treat the Bayes factor as an alternative option for p-value (ranking the Bayes factor for all the units from the biggest to the smallest) and then calculate the power and FDR like what we usually do with the p-value?

Best

**Comment by stephens999** [comment]
yes, to plot a curve like this all you need is a ranking. p values and BFs
both provide a ranking,
and the curves are just quantifying how effective that ranking is from
distinguishing the effects
and the non-effects. You could do this for any method
that provides a ranking, it doesn't have to be a p value or a BF....
